apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -715,7 +721,18 @@ def _build_tiledb_schema(
             )
 
     dims = []
-    for index_column_name, slot_domain in zip(index_column_names, domain):
+    for index_column_name, _slot_domain in zip(index_column_names, domain):
+        # TODO: this can be cleaned up a bit in Python ≥3.10 (see note on ``Domain`` type above)
+        if _slot_domain is not None:
+            if not isinstance(_slot_domain, tuple):
+                if len(_slot_domain) != 2:
+                    raise ValueError(
+                        f"Domain slots must be tuples or lists of length 2; received {len(_slot_domain)}-list {_slot_domain}"
+                    )
+            slot_domain = _slot_domain[0], _slot_domain[1]
+        else:
+            slot_domain = None
```

Comment:
I don’t follow this check. Why is the length only being checked if it’s a tuple? It seems like this could be reduced to:

```python
if slot_domain is not None:
  slot_domain = tuple(slot_domain)
  if len(slot_domain) != 2:
    raise ...
```

which nicely folds all non-tuple sequences down to tuples and ensures that they have the appropriate length.